### ISSUE TITLE
Reorder File Sequence

### ISSUE THREAD

**Issue by Nasidiqi** [affans/yorkuthesis#2]
also how can i re order the file in front matter like Abstract, acknowledgement preface shoudl come before TOC.

**Comment by affans** [comment]
Isn't that currently the case? The title page, abstract, acknowledgements all come before the TOC with the current code. In any case, if you want to change the order, look at line 118 in the `cls` file. See code below and my comment therein.
```
\def\makefrontmatter{%
  \pagestyle {empty}
  \pagenumbering {roman}
  \setcounter {page} {1}
  \maketitlepage
  %%\pagestyle {empty}  \makecopyrightpage
  %%\pagestyle {empty}  \makeauthenticitypage  
  \@ifdefined {@abstractfile}{\makeabstract}
  \pagestyle {fancy}
  \@ifdefined {@dedicationfile}{\makededication}
  \@ifdefined {@acknowledgementsfile}{\makeacknowledgements}
  \tableofcontents   %% MOVE THIS LINE UP AND DOWN TO MOVE THE TABLE OF CONTENTS.
  \clearpage
  \ifthenelse{\boolean{hastables}}{\listoftables}{}
  \clearpage
  \ifthenelse{\boolean{hasfigures}}{\listoffigures}{}
  \@ifdefined{@prefacefile}{\makepreface}
  \@ifdefined{@abbreviationsfile}{\makeabbreviations}
  \newpage
  \pagenumbering {arabic}
  \setcounter {page}{1}}
```

**Comment by Nasidiqi** [comment]
NO, TOC is appearing on top. Can you help me to format this? somewhere live i can share thesis with u ?

**Comment by Nasidiqi** [comment]
Title page is not appearing. this is my  tex file.

\documentclass{york-thesis}

%========== basic required packages  =================
\usepackage[tracking]{microtype}
\usepackage[T1]{fontenc}  
\usepackage[utf8]{inputenc} 
\usepackage{pdfpages}

%========== citation setup  =================
%\usepackage[backend=biber, style=numeric, sorting=none, natbib=true, backref=false, backrefstyle=all+, hyperref=false]{biblatex}
%\addbibresource{./%FILENAME%.bib} % enter with .bib extension


%========== thesis setup  =================
% \setboolean{masters}{false}      % set true for a Master's thesis; % false for a PhD dissertation
% \setboolean{hasfigures}{true}  % set true if you have figures
% \setboolean{hastables}{true}   % set true of you have tables

\title{A natural language question answering system for exploring online conversations}       
\author{Nadia Ashfaq Siddiqui}   % this is your name exactly as you want it to appear everywhere in your work.
% \department{Mathematics and Statistics}   % this is the name of the programme in which you are attempting your degree
% \masterof{Science}        % this is the type of Master's degree you are attempting. If the boolean masters is false, this has no effect.
% \degreename{Doctor}       % put the actual name of the degree you are attempting {Doctor or Masters}
% \date{September 2018}     % this is the month and year of defence

% Your committee members list has to be an enumerated list.
%
% \committeememberslist{
%   \begin{enumerate}
%     \item Member A
%     \item Member B
%     \item Member C
%   \end{enumerate}}

%========== Front Matter Files  =================
%\input{Title}
\abstractfile{Abstract.tex}
%\dedicationfile{dedication.tex}
\acknowledgementsfile{Acknowledgement.tex}
\prefacefile{Preface.tex}

%\abbreviationsfile{abbreviations.tex}
%
\toc

\begin{document}
\makefrontmatter
    % main text follows
    \include{introduction}
    \include{Problem}
    \include{literature_review}
    \include{proposed_approaches}
    \include{Interface}
    \include{UserStudy}
    \include{FutureWork}
    %% include your chapters here. 
    %% include your appendix here.
    \include{appendix-a}

\end{document}

**Comment by affans** [comment]
Get rid of the `\toc`. You don't need it, the `\makefrontmatter` command already orders and puts in the table of contents for you.

**Comment by Nasidiqi** [comment]
it does not work as it is suppose to

**Comment by affans** [comment]
Could you paste your `.tex` file? Please use ` ``` ` to surround your code to make it readable, something like 

```
% ```
my code
% ``` 
```
(without the `%`)

**Comment by Nasidiqi** [comment]
````````````````````````
\documentclass{york-thesis}

%========== basic required packages  =================
\usepackage[tracking]{microtype}
\usepackage[T1]{fontenc}  
\usepackage[utf8]{inputenc} 
\usepackage{pdfpages}

%========== citation setup  =================
%\usepackage[backend=biber, style=numeric, sorting=none, natbib=true, backref=false, backrefstyle=all+, hyperref=false]{biblatex}
%\addbibresource{./%FILENAME%.bib} % enter with .bib extension


%========== thesis setup  =================
% \setboolean{masters}{false}      % set true for a Master's thesis; % false for a PhD dissertation
% \setboolean{hasfigures}{true}  % set true if you have figures
% \setboolean{hastables}{true}   % set true of you have tables

\title{A natural language question answering system for exploring online conversations}       
\author{Nadia Ashfaq Siddiqui}   % this is your name exactly as you want it to appear everywhere in your work.
% \department{Mathematics and Statistics}   % this is the name of the programme in which you are attempting your degree
% \masterof{Science}        % this is the type of Master's degree you are attempting. If the boolean masters is false, this has no effect.
% \degreename{Doctor}       % put the actual name of the degree you are attempting {Doctor or Masters}
% \date{September 2018}     % this is the month and year of defence

% Your committee members list has to be an enumerated list.
%
% \committeememberslist{
%   \begin{enumerate}
%     \item Member A
%     \item Member B
%     \item Member C
%   \end{enumerate}}

%========== Front Matter Files  =================
%\input{Title}
\abstractfile{Abstract.tex}
%\dedicationfile{dedication.tex}
\acknowledgementsfile{Acknowledgement.tex}
\prefacefile{Preface.tex}

%\abbreviationsfile{abbreviations.tex}
%

\begin{document}
\makefrontmatter
    % main text follows
    \include{introduction}
    \include{Problem}
    \include{literature_review}
    \include{proposed_approaches}
    \include{Interface}
    \include{UserStudy}
    \include{FutureWork}
    %% include your chapters here. 
    %% include your appendix here.
    \include{appendix-a}

\end{document}


````````````

**Comment by Nasidiqi** [comment]
it starts with Abstract and is not creating title page. 


Aso it is not adding these in this seq in Tabel of content.

- Abstract
- Acknowledgement
- Preface
- table of content

I have spent many hours but cant figure out.

**Comment by affans** [comment]
Sorry you are having trouble. Your file looks correct to me. A couple of questions 
1) Have you create the files `Abstract.tex` , `Acknowledgement.tex` and `Preface.tex`? They should be in the same folder. 
2) Does your code compile properly? (I would recommend deleting all latex-produced files (except the `.tex` and `.cls`) and recompile again.

**Comment by Nasidiqi** [comment]
Yes, i have and they are in same folder.

What about teh title page? It does not display....

**Comment by affans** [comment]
Can you attach the pdf you create?

**Comment by Nasidiqi** [comment]
[ConViaQA_Thesis_Doc__Latest_with_track_changes_ (1).pdf](https://github.com/affans/yorkuthesis/files/5895369/ConViaQA_Thesis_Doc__Latest_with_track_changes_.1.pdf)

**Comment by Nasidiqi** [comment]
[ConViaQA_Thesis_Doc__Latest_with_track_changes___Copy_.pdf](https://github.com/affans/yorkuthesis/files/5895404/ConViaQA_Thesis_Doc__Latest_with_track_changes___Copy_.pdf)

**Comment by Nasidiqi** [comment]
Please help me .....................its not working...

**Comment by affans** [comment]
Okay, so now I am a bit confused at what you are missing or what you intend to do? It seems to me that your PDF (the first file you attached) does have a title page. It's the first page of the PDF. It seems like the Abstract, Acknowledgements, and Preface are also there, **before** the table of contents like you wanted (and how FGS wants). The TOC also shows the correct pages including the links to abstract, acknowledgements, and preface. 

Take a look at FGS requirements https://gradstudies.yorku.ca/current-students/thesis-dissertation/thesis/

I am not sure what you are missing here?

Edit: Why is there such a drastic change between the first and second file? I would not suggest editing the `.cls` if you don't need to, or don't have the expertise yet. The current organization matches that of FGS and they are fairly strict about it from what I remember.

**Comment by Nasidiqi** [comment]
Did you look at  pdf? IT just updated and displayed these contents. but its not complete it is not showing rest of thesis content/ all chapter, bibliography.

that is from another template i submitted initially and now i need to work on FGS feedback. I have never worked in latex and it is bit complex.

**Comment by affans** [comment]
Yes, I am looking at the first file you attached. It has 78 pages, with the title page, and all the other front matter files. It also has all the chapters, and those chapters show up in the TOC as well. 

You can send me a zoom/google hangout link, I have about 30 minutes I can use to help you.

**Comment by Nasidiqi** [comment]
what email address to send u zoom link ?

**Comment by affans** [comment]
affan.shoukat@yale.edu

**Comment by Nasidiqi** [comment]
https://zoom.us/j/95848029891?pwd=OHp0OHhPd1VHSmI3Tk1JNXZESUJGUT09
